Source/Embeddings.Store.MongoDB/EmbeddingsConnection.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Dolittle.Runtime.Lifecycle;
+using MongoDB.Driver;
+
+namespace Dolittle.Runtime.Embeddings.Store.MongoDB
+{
+    /// <summary>
+    /// Represents a connection to the MongoDB Embeddings database.
+    /// </summary>
+    [SingletonPerTenant]
+    public class EmbeddingsConnection : IEmbeddingsConnection
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventStoreConnection"/> class with default collections.
+        /// </summary>
+        /// <param name="connection">A connection to the MongoDB database.</param>
+        protected EmbeddingsConnection(DatabaseConnection connection)
+        {
+            MongoClient = connection.MongoClient;
+            Database = connection.Database;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="IMongoClient"/> configured for the MongoDB database.
+        /// </summary>
+        protected IMongoClient MongoClient { get; }
+
+        /// <summary>
+        /// Gets the <see cref="IMongoDatabase" />.
+        /// </summary>
+        protected IMongoDatabase Database { get; }
+
+        /// <summary>
+        /// Starts a client session.
+        /// </summary>
+        /// <param name="options">The <see cref="ClientSessionOptions" />.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken" />.</param>
+        /// <returns>The <see cref="IClientSessionHandle" />.</returns>
+        public IClientSessionHandle StartSession(ClientSessionOptions options = default, CancellationToken cancellationToken = default) => MongoClient.StartSession(options, cancellationToken);
```

Comment:
I was thinking about this earlier too, the only reason why I see it being a bad thing is because Codacy tells us so ( and we haven't configured it to not complain). This is just copy paste code from the other `XConnection` classes that deal with this stuff, and I think that it's easier to have them all be mostly the same for now so that when we do give them all a proper unified mongo abstraction, it's gonna be a bit simpler to change them.